### ISSUE TITLE
如何保存PB

### ISSUE THREAD

**Issue by LanWong1** [chizhanyuefeng/DB-tf#6]
您好，请问这个如何讲chechpoint保存为pb文件？我在tensorboard上并找不到输出节点是哪些，所以我就转换失败了。还请您赐教。谢谢！

**Comment by chizhanyuefeng** [comment]
显示时候下面的代码，稍后我会加到repo中。
`def ckpt2pb(ckptpath):

    os.environ['CUDA_VISIBLE_DEVICES'] = '0'
    tf.reset_default_graph()
    _input_images = tf.placeholder(tf.float32, shape=[None, None, None, 3], name='input_images')
    global_step = tf.get_variable('global_step', [], initializer=tf.constant_initializer(0), trainable=False)

    _binarize_map, _threshold_map, _thresh_binary = model.model(_input_images, is_training=False)

    variable_averages = tf.train.ExponentialMovingAverage(0.997, global_step)
    saver = tf.train.Saver(variable_averages.variables_to_restore())
    gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=0.5)
    gpu_config = tf.ConfigProto(log_device_placement=False, gpu_options=gpu_options, allow_soft_placement=True)
    sess = tf.Session(config=gpu_config)
    saver.restore(sess, ckptpath)

    from tensorflow.python.framework import graph_util
    constant_graph = graph_util.convert_variables_to_constants(
        sess,
        sess.graph_def,
        ['feature_fusion/binarize_branch/Conv2d_transpose_1/Sigmoid'])

    with tf.gfile.FastGFile('db.pb', mode='wb') as f:
        f.write(constant_graph.SerializeToString())`

**Comment by chizhanyuefeng** [comment]
见lib/utils.py

**Comment by LanWong1** [comment]
谢谢！我本来是用meta来加载模型 但是找不到这个node。不知道为什么


祝好！
王毅
在2019年12月27日 18:12，赤战约风 写道：

见lib/utils.py
